packages/react-instantsearch-core/src/core/__tests__/createInstantSearchManager.js

```diff
@@ -613,6 +624,134 @@ describe('createInstantSearchManager', () => {
         },
       ]);
     });
+
+    it('expects widgets main parameters and derived parameters to be correctly calculated within a multi index context', () => {
+      const wrapper = mount(
+        <InstantSearch indexName="index1" searchClient={createSearchClient()}>
+          <Index indexName="bestbuy" />
+          <Index indexName="instant_search" />
+
+          <Index indexId="instant_search_apple" indexName="instant_search">
+            <Configure filters="brand:Apple" />
+          </Index>
+
+          <Index indexId="instant_search_samsung" indexName="instant_search">
+            <Configure filters="brand:Samsung" />
+          </Index>
+
+          <Index indexId="instant_search_microsoft" indexName="instant_search">
+            <Configure filters="brand:Microsoft" />
+          </Index>
+        </InstantSearch>
+      );
+
+      const {
+        mainParameters,
+        derivedParameters,
+      } = wrapper.instance().aisManager.getSearchParameters();
+
+      expect(mainParameters).toEqual(
+        expect.objectContaining({
+          index: 'index1',
+        })
+      );
+
+      expect(derivedParameters).toHaveLength(5);
+
+      expect(derivedParameters).toEqual(
+        expect.arrayContaining([
```

Comment:
I don't think this part is useful, we can directly use `expect.objectContaining`.